framework/src/media/audio/audio_manager.c

```diff
@@ -1253,58 +1256,50 @@ uint32_t get_output_bytes_to_frame(unsigned int bytes)
 
 audio_manager_result_t get_max_audio_volume(uint8_t *volume)
 {
-	// ToDo: Add logics to consider 'decibel'.
-	if (volume == NULL) {
-		return AUDIO_MANAGER_INVALID_PARAM;
-	}
-
 	*volume = AUDIO_DEVICE_MAX_VOLUME;
 
 	return AUDIO_MANAGER_SUCCESS;
 }
 
 audio_manager_result_t get_input_audio_gain(uint8_t *gain)
 {
-	int ret = AUDIO_MANAGER_SUCCESS;
+	audio_manager_result_t ret = AUDIO_MANAGER_SUCCESS;
 	audio_card_info_t *card;
-	
+
 	if (gain == NULL) {
 		return AUDIO_MANAGER_INVALID_PARAM;
 	}
 	ret = get_audio_volume(INPUT);
 	if (ret != AUDIO_MANAGER_SUCCESS) {
 		return ret;
-	}	
+	}
 
 	card = &g_audio_in_cards[g_actual_audio_in_card_id];
 
-	medvdbg("ret = %d, Max volume: %d, Current Volume : %d card id : %d device id : %d\n", ret, card->config[card->device_id].max_volume, \
-		card->config[card->device_id].volume, g_actual_audio_in_card_id, card->device_id);
-
+	*gain = card->config[card->device_id].volume * AUDIO_DEVICE_MAX_VOLUME / card->config[card->device_id].max_volume;
```

Comment:
we calculate volume in get_audio_volume already,  I think this is wrong.
if card->config.max_volume is 0xBF & card->config.volume is 64, volume will calculated as 1D
Then 1D * 10 / (BF - (BF % 10 = 1) = 1D * 10 / 190 = 1.52

Reply:
Is there any specific reason? you made duplicated code.
If you want separate it, there should be reason (for example, different MAX values)